src/weblets/tf_presearch.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+ <weblet-layout>
+  <template #title>Deploy a Presearch Instance</template>
+    <template #subtitle
+      >Presearch is a community-powered, decentralized search engine that provides better results while protecting your privacy and rewarding you when you search. This weblet deploys a Presearch node. Presearch Nodes are used to process user search requests, and node operators earn Presearch PRE tokens for joining and supporting the network. 
+      <a
+        class="app-link"
+        href="https://manual.grid.tf/weblets/weblets_presearch.html"
+        target="_blank"
+      >
+        Quick start documentation
+      </a>
+    </template>
+
+
+    <d-tabs 
+      :tabs="[
+        { title: 'Base', value: 'base' },
+        { title: 'Restore', value: 'restore' }
+      ]"
+    >
+      <template #base>
+        <v-text-field label="Name" v-model="name" />
+
+        <password-input-wrapper>
+          <template #default="{ props }">
+            <v-text-field label="Presearch Registeration Code" v-bind="props" v-model="code" />
+          </template>
+        </password-input-wrapper>
+
+
+        <v-switch color="primary" inset label="Public IPv4" v-model="ipv4" />
+        <v-switch color="primary" inset label="Planetary Network" v-model="planetary" />
+        <SelectFarm
+          :filters="{
+            cpu,
+            memory,
+            ssd: rootFsSize
+          }"
+          v-model="farm"
+        />
+      </template>
+
+
+      <template #restore>
+        <v-textarea label="Private Presearch Restore Key" v-model="privateRestoreKey" no-resize :spellcheck="false" />
+        <v-textarea label="Public Presearch Restore Key" v-model="publicRestoreKey" no-resize :spellcheck="false" />
+      </template>
+
+    </d-tabs>
+    <template #footer-actions>
+      <v-btn color="primary" variant="tonal" :loading="loading" :disabled="loading" @click="deploy">
+        Deploy
+      </v-btn>
+    </template>
+ </weblet-layout>
+</template>
+
+<script lang='ts' setup>
+import { ref, type Ref } from 'vue'
+import { generateString } from 'grid3_client'
+import type { Farm } from '../types'
+import { deployVM, type Env } from '../utils/deploy_vm'
+import { useProfileManager } from '../stores'
+import { getGrid } from '../utils/grid'
+import rootFs from "../utils/rootFs"
+
+
+const profileManager = useProfileManager()
+const name = ref('PS' + generateString(8))
+const code = ref() as Ref<string>
+const ipv4 = ref(false)
+const planetary = ref(true)
+const cpu = ref(4)
+const memory = ref(8192)
+const rootFsSize = ref(rootFs(cpu.value, memory.value)) as Ref<number>
```

Comment:
```suggestion
const cpu = 4
const memory = 8192
const rootFsSize = rootFs(cpu.value, memory.value)
```

Please turn them into regular constants and there is no point for making them ref